pipenv/utils.py

```diff
@@ -241,53 +243,40 @@ class PipCommand(basecommand.Command):
         cleanup_req_dir = True
     for dep in deps:
         if dep:
-            if dep.startswith('-e '):
-                constraint = req.InstallRequirement.from_editable(
-                    dep[len('-e '):]
-                )
-            else:
-                fd, t = tempfile.mkstemp(
-                    prefix='pipenv-', suffix='-requirement.txt', dir=req_dir.name
-                )
-                with os.fdopen(fd, 'w') as f:
-                    f.write(dep)
-                constraint = [
-                    c for c in req.parse_requirements(t, session=pip_requests)
-                ][
-                    0
-                ]
-            # extra_constraints = []
+            url = None
             if ' -i ' in dep:
-                index_lookup[constraint.name] = project.get_source(
-                    url=dep.split(' -i ')[1]
-                ).get(
-                    'name'
-                )
-            if constraint.markers:
-                markers_lookup[constraint.name] = str(
-                    constraint.markers
-                ).replace(
-                    '"', "'"
-                )
-            constraints.append(constraint)
+                dep, url = dep.split(' -i ')
+            req = Requirement.from_line(dep)
+            _line = req.as_line()
+            constraints.append(_line)
```

Comment:
What does this `from_line` → `to_line` combination do? Can’t `dep` be added directly into `constraints`?